internal/provider/provider.go

```diff
@@ -177,26 +172,63 @@ func (p *PrefectProvider) Configure(ctx context.Context, req provider.ConfigureR
 	} else if accountIDEnvVar, ok := os.LookupEnv("PREFECT_CLOUD_ACCOUNT_ID"); ok {
 		accountID, err = uuid.Parse(accountIDEnvVar)
 		if err != nil {
-			resp.Diagnostics.AddAttributeWarning(
+			resp.Diagnostics.AddAttributeError(
 				path.Root("account_id"),
 				"Invalid Prefect Account ID defined in PREFECT_CLOUD_ACCOUNT_ID ",
 				fmt.Sprintf("The PREFECT_CLOUD_ACCOUNT_ID value %q is not a valid UUID: %s", accountIDEnvVar, err),
 			)
+
+			return
+		}
+	} else if urlContainsIDs(endpoint) {
+		aID, err := getAccountIDFromPath(endpointURL.Path)
+		if err != nil {
+			resp.Diagnostics.AddAttributeError(
+				path.Root("account_id"),
+				"Invalid Prefect Account ID defined in PREFECT_API_URL ",
+				fmt.Sprintf("The PREFECT_API_URL contains a workspace value is not a valid UUID: %s", err),
+			)
+
+			return
+		}
+
+		accountID = aID
+	}
+
+	// Extract the Workspace ID from configuration or environment variable.
+	// If the ID is set to an invalid UUID, emit an error.
+	var workspaceID uuid.UUID
+	if !config.WorkspaceID.IsNull() {
+		workspaceID = config.WorkspaceID.ValueUUID()
+	} else if urlContainsIDs(endpoint) {
+		wID, err := getWorkspaceIDFromPath(endpointURL.Path)
+		if err != nil {
+			resp.Diagnostics.AddAttributeError(
+				path.Root("workspace_id"),
+				"Invalid Prefect Workspace ID defined in PREFECT_API_URL ",
+				fmt.Sprintf("The PREFECT_API_URL contains a workspace value is not a valid UUID: %s", err),
+			)
+
+			return
```

Comment:
The logic here works, but pre-commit fairly pointed out this is getting a bit complex. Will work on simplifying out the logic here for better maintainability and legibility.

Reply:
I touched up this method a bit, but reality is there's just a lot of work to do here. I could split it off to some helper functions, but honestly not sure that really improves maintainability. Open to any feedback here.